### PR TITLE
get parent constraints

### PR DESCRIPTION
**Pull request by arekkubaczkowski**

### CODE REVIEW

File: src/Form/FormDataConstraintFinder.php [hunk 1]
```diff
@@ -155,16 +156,43 @@ private function findPropertyConstraints(ClassMetadata $metadata, $propertyName,
...
+     *
+     * @param MemberMetadata $propertyMetadata
+     * @param ClassMetadata $metadata
+     * @return array
+     */
+    protected function addParentValidations(MemberMetadata $propertyMetadata, ClassMetadata $metadata) {
+        $constraints = [];
+
+        foreach ($propertyMetadata->getConstraints() as $constraint) {
+            if (get_class($constraint) == ValidateByParent::class) {
```
Comment (ikselll) [inline src/Form/FormDataConstraintFinder.php]: Please use strcmp to compare stings and change condition to oposite and use continue inside - less nesting.

File: src/Validator/ValidateByParent.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * validateByParent
+ *
+ * @author Arkadiusz Kubaczkowski <arkadiusz.kubaczkowski@pixers.pl>
+ */
+class ValidateByParent extends Constraint
+{
+    protected $constraint;
```
Comment (ikselll) [inline src/Validator/ValidateByParent.php]: @var type?

File: src/Validator/ValidateByParent.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    public function getConstraintName() {
```
Comment (ikselll) [inline src/Validator/ValidateByParent.php]: docs

File: src/Validator/ValidateByParentValidator.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * ValidateByParentValidator
+ *
+ * @author Arkadiusz Kubaczkowski <arkadiusz.kubaczkowski@pixers.pl>
+ */
+class ValidateByParentValidator extends ConstraintValidator
+{
+   public function validate($value, Constraint $constraint) {}
```
Comment (ikselll) [inline src/Validator/ValidateByParentValidator.php:20]: docs

**Review by ikselll — CHANGES_REQUESTED** [review]

File: src/Form/FormDataConstraintFinder.php [hunk 5]
```diff
@@ -155,16 +156,45 @@ private function findPropertyConstraints(ClassMetadata $metadata, $propertyName,
...
     }
 
     /**
+     * Gets parent constraints.
+     *
+     * @param MemberMetadata $propertyMetadata
+     * @param ClassMetadata $metadata
+     * @return array
+     */
+    protected function addParentValidations(MemberMetadata $propertyMetadata, ClassMetadata $metadata) {
```
Comment (michallkanak) [inline src/Form/FormDataConstraintFinder.php]: psr - brackets

File: src/Validator/ValidateByParent.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+{
+    /**
+     * @var string
+     */
+    protected $constraintName;
+
+    /**
+     * @return string
+     */
+    public function getConstraintName() {
```
Comment (michallkanak) [inline src/Validator/ValidateByParent.php]: psr brackets

File: src/Validator/ValidateByParentValidator.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+ * @author Arkadiusz Kubaczkowski <arkadiusz.kubaczkowski@pixers.pl>
+ */
+class ValidateByParentValidator extends ConstraintValidator
+{
+    /**
+     * @param string $value
+     * @param Constraint $constraint
+     * @return void
+     */
+   public function validate($value, Constraint $constraint) {}
```
Comment (michallkanak) [inline src/Validator/ValidateByParentValidator.php:20]: framework need this class to work?

File: src/Form/FormDataConstraintFinder.php [hunk 8]
```diff
@@ -155,16 +156,45 @@ private function findPropertyConstraints(ClassMetadata $metadata, $propertyName,
...
+        $constraints = [];
+
+        foreach ($propertyMetadata->getConstraints() as $constraint) {
+            if (strcmp(get_class($constraint), ValidateByParent::class) !== 0) {
+                continue;
+            }
+
+            foreach ($metadata->getConstraints() as $parentConstraint) {
+                $shortName = (new \ReflectionClass($parentConstraint))->getShortName();
```
Comment (michallkanak) [inline src/Form/FormDataConstraintFinder.php]: $shortName -> $parentConstraintName

File: src/Form/FormDataConstraintFinder.php [hunk 9]
```diff
@@ -155,16 +156,45 @@ private function findPropertyConstraints(ClassMetadata $metadata, $propertyName,
                 continue;
             }
 
+            $constraintsWithParents = $this->addParentValidations($propertyMetadata, $metadata);
```
Comment (michallkanak) [inline src/Form/FormDataConstraintFinder.php]: constraintsWithParents -> constraintsFromParent ?

File: src/Validator/ValidateByParentValidator.php (hunk 7 above)
Comment (arekkubaczkowski) [reply src/Validator/ValidateByParentValidator.php:20]: Yes, its needed.

File: src/Form/FormDataConstraintFinder.php [hunk 10]
```diff
@@ -155,16 +156,46 @@ private function findPropertyConstraints(ClassMetadata $metadata, $propertyName,
...
+        $constraints = [];
+
+        foreach ($propertyMetadata->getConstraints() as $constraint) {
+            if (strcmp(get_class($constraint), ValidateByParent::class) !== 0) {
+                continue;
+            }
+
+            foreach ($metadata->getConstraints() as $parentConstraint) {
+                $parentConstraintName = (new \ReflectionClass($parentConstraint))->getShortName();
+                if ($parentConstraintName = $constraint->getConstraintName()) {
```
Comment (ikselll) [inline src/Form/FormDataConstraintFinder.php]: When this condition is true @arekkubaczkowski?

**Review by ikselll — CHANGES_REQUESTED** [review]

**Pull request merged**